src/credentials/CHIPCert.h

```diff
@@ -623,14 +620,11 @@ CHIP_ERROR DecodeChipDN(chip::TLV::TLVReader & reader, ChipDN & dn);
  * @brief Convert standard X.509 certificate to CHIP certificate.
  *
  * @param x509Cert        CHIP X.509 DER encoded certificate.
- * @param chipCertBuf     Buffer to store converted certificate in CHIP format.
- * @param chipCertBufSize The size of the buffer to store converted certificate.
- * @param chipCertLen     The length of the converted certificate.
+ * @param chipCert        Buffer to store converted certificate in CHIP format.
  *
  * @return Returns a CHIP_ERROR on error, CHIP_NO_ERROR otherwise
  **/
-CHIP_ERROR ConvertX509CertToChipCert(const ByteSpan x509Cert, uint8_t * chipCertBuf, uint32_t chipCertBufSize,
-                                     uint32_t & chipCertLen);
+CHIP_ERROR ConvertX509CertToChipCert(const ByteSpan x509Cert, MutableByteSpan & chipCert);
```

Comment:
A bytespan is already const

Reply:
A ByteSpan is a span over const bytes, but the object itself is not const.

That said, passing it by value means that mutating it (e.g. changing the size, or reassigning it to point to a different buffer altogether) won't affect the caller anyway... the benefit of making it const at that point is to make it clear that you should not do these things.  That said, if it's const anyway, worth checking whether const by value or const ref leads to smaller code or whether it's all the same in the end.

Reply:
@tcarmelveilleux ?